src/base/static/state/ducks/map.js

```diff
@@ -112,14 +156,92 @@ export default function reducer(state = INITIAL_STATE, action) {
           ...action.payload,
         },
       };
-    case SET_LAYER_STATUS:
+    case SET_BASEMAP:
+      newStatuses = {
+        [action.payload]: {
+          ...state.layerStatuses[action.payload],
+          isVisible: true,
+          loadStatus: "loading",
+        },
+      };
+
+      // If a previous basemap was already visible, switch it off.
+      if (state.visibleBasemapId) {
+        newStatuses[state.visibleBasemapId] = {
+          ...state.layerStatuses[state.visibleBasemapId],
+          isVisible: false,
+        };
+      }
+
+      return {
+        ...state,
+        visibleBasemapId: action.payload,
+        layerStatuses: {
+          ...state.layerStatuses,
+          ...newStatuses,
+        },
+      };
+    case INIT_LAYER:
+      newStatuses = action.payload.reduce((memo, loadStatus) => {
+        return {
+          ...memo,
+          [loadStatus.id]: {
+            ...loadStatus,
+          },
+        };
+      }, {});
+
       return {
         ...state,
         layerStatuses: {
           ...state.layerStatuses,
-          [action.payload.layerId]: {
-            ...state.layerStatuses[action.payload.layerId],
-            ...action.payload.layerStatus,
+          ...newStatuses,
+        },
+      };
+    case UPDATE_LAYERS_VISIBLE:
+      newStatuses = action.payload.reduce((memo, layerId) => {
+        return {
+          ...memo,
+          [layerId]: {
+            ...state.layerStatuses[layerId],
+            isVisible: true,
+          },
+        };
+      }, {});
+
+      return {
+        ...state,
+        layerStatuses: {
+          ...state.layerStatuses,
+          ...newStatuses,
+        },
+      };
+    case UPDATE_LAYERS_HIDDEN:
+      newStatuses = action.payload.reduce((memo, layerId) => {
+        return {
+          ...memo,
+          [layerId]: {
+            ...state.layerStatuses[layerId],
+            isVisible: false,
+          },
+        };
+      }, {});
+
+      return {
+        ...state,
+        layerStatuses: {
+          ...state.layerStatuses,
+          ...newStatuses,
+        },
+      };
+    case SET_LAYER_LOAD_STATUS:
+      return {
+        ...state,
+        layerStatuses: {
+          ...state.layerStatuses,
+          [action.payload.id]: {
+            ...state.layerStatuses[action.payload.id],
+            ...action.payload,
```

Comment:
I think we should replace this line with:

```javascript
loadStatus: action.payload.loadStatus,
```

This would make it clear that we are only setting the `loadStatus` attribute of the LayerStatus.